Bug 1894423 - Remove unused ExclusiveData move constructor. r=spidermonkey-reviewers,jonco

Because the constructor is actually not used, the compiler used to not
complain about it being broken. Recent changes on clang trunk made it
catch this problem without the constructor being used.

As Mutex doesn't have a move constructor, it's also not only a matter of
adding the missing underscore to lock.

As the constructor is never used, just remove it.

Differential Revision: https://phabricator.services.mozilla.com/D209108

Upstream-Status: Backport [https://hg.mozilla.org/mozilla-central/rev/223087fdc29f]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- a/js/src/threading/ExclusiveData.h
+++ b/js/src/threading/ExclusiveData.h
@@ -109,11 +109,6 @@ class ExclusiveData {
   explicit ExclusiveData(const MutexId& id, Args&&... args)
       : lock_(id), value_(std::forward<Args>(args)...) {}
 
-  ExclusiveData(ExclusiveData&& rhs)
-      : lock_(std::move(rhs.lock)), value_(std::move(rhs.value_)) {
-    MOZ_ASSERT(&rhs != this, "self-move disallowed!");
-  }
-
   ExclusiveData& operator=(ExclusiveData&& rhs) {
     this->~ExclusiveData();
     new (mozilla::KnownNotNull, this) ExclusiveData(std::move(rhs));
